library/Rules/Uuid.php

```diff
@@ -13,19 +13,78 @@
 
 namespace Respect\Validation\Rules;
 
+use Respect\Validation\Exceptions\ComponentException;
 use function is_scalar;
 use function preg_match;
 
+/**
+ * This class validates UUIDs.
+ *
+ * Henrique Moody <henriquemoody@gmail.com>
+ * Michael Weimann <mail@michael-weimann.eu>
+ */
 class Uuid extends AbstractRule
 {
-    private const PATTERN = '/^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i';
+    public const VERSION_ALL = '[1-5]';
+    public const VERSION_1 = '1';
+    public const VERSION_2 = '2';
+    public const VERSION_3 = '3';
+    public const VERSION_4 = '4';
+    public const VERSION_5 = '5';
+
+    public const VERSIONS = [
+        self::VERSION_ALL,
+        self::VERSION_1,
+        self::VERSION_2,
+        self::VERSION_3,
+        self::VERSION_4,
+        self::VERSION_5,
+    ];
 
+    /**
+     * Uuid regex pattern with sprint version placeholder.
+     */
+    private const PATTERN = '/^[0-9a-f]{8}-[0-9a-f]{4}-%s[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i';
+
+    /**
+     * The UUID version to validate for.
+     *
+     * string
+     */
+    private $version;
+
+    /**
+     * Uuid constructor.
+     *
+     * @param string $version use one of the Uuid class constants
+     *
+     * @throws ComponentException
+     */
+    public function __construct(string $version = self::VERSION_ALL)
```

Comment:
I think that if you don't wanna do a specific version you just don't pass the version.